docs/excel/excel-add-ins-pivots.md

```diff
@@ -0,0 +1,243 @@
+
+> [!NOTE]
+> These samples use APIs currently available only in public preview (beta). To run these samples, you must use the beta library of the Office.js CDN: https://appsforoffice.microsoft.com/lib/beta/hosted/office.js.
+
+## Hierarchies
+
+PivotTables are organized based on four hierarchy categories: row, column, data, and filter. The following data describing fruit sales from various farms will be used in throughout this article.
+
+![A collection of fruit sales of different types from different farms.](../images/excel-pivots-raw-data.png)
+
+This data has five hierarchies: **Farms**, **Type**, **Classification**, **Crates Sold at Farm**, and **Crates Sold Wholesale**. Each hierarchy can only exist in one of the four categories. If **Type** is added to column hierarchies and then added to row hierarchies, it only remains in the latter.
+
+Row and column hierarchies define how data will be grouped. For example, having a row hierarchy of **Farms** groups together all data sets from the same farm. The choice between row and column hierarchy defines the orientation of the PivotTable.
+
+Data hierarchies are the values to be aggregated based on the row and column hierarchies. A PivotTable with a row hierarchy of **Farms** and a data hierarchy of **Crates Sold Wholesale** shows the sum total (by default) of all the different fruits for each farm.
+
+Filter hierarchies include or exclude data from the pivot based on values within that filtered type. A filter hierarchy of **Classification** with the type **Organic** selected only shows data for organic fruit.
+
+Here is the farm data again, alongside a PivotTable. The PivotTable is using **Farm** and **Type** as the row hierarchies, **Crates Sold at Farm** and **Crates Sold Wholesale** as the data hierarchies (with the default aggregation function of sum), and **Classification** as a filter hierarchy (with **Organic** selected). 
+
+![A selection of fruit sales data next to a PivotTable with row, data, and filter hierarchies.](../images/excel-pivot-table-and-data.png)
+
+This PivotTable could be generated through the JavaScript API or through the Excel UI. Both options allow for further manipulation through add-ins.
+
+## Create a PivotTable
+
+PivotTables need a name, source, and destination. The source can be a `Range`, `string`, or `Table`. The destination can either be a `Range` or `string`. 
+The following samples show various PivotTable creation techniques. All three samples feature a PivotTable named **Farm Sales** created on a worksheet called **PivotWorksheet** at cell **A2**. Its data comes from the worksheet **DataWorksheet** across the range **A1:E21**. 
+
+### Create a PivotTable with strings
+```ts
+await Excel.run(async (context) => {
+	context.workbook.worksheets.getActiveWorksheet()
+		.pivotTables.add("Farm Sales", "DataWorksheet!A1:E21", "PivotWorksheet!A2");
```

Comment:
add a comment explaining each of the parameters (valid for all samples!) usually devs only look t the code